google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java

```diff
@@ -83,6 +85,57 @@ static FieldPath fromDotSeparatedString(String field) {
     return empty().append(field);
   }
 
+  /** Creates a {@code FieldPath} from a server-encoded field path. */
+  public static FieldPath fromServerFormat(String path) {
```

Comment:
Java sdk should have handled field name with dot before without this function. I am checking if there are any other ways to do this.

Reply:
I think the SDK didn't need it before.